### DISCUSSION TITLE
url_spec correct for a vCenter, but wrong for other.

### DISCUSSION THREAD

**Discussion by elungu3** [vmware/pyvmomi-community-samples#738] [Q&A] [answered]
Hello @prziborowski 

I am using you "upload_file_to_datastore.py" method and I used it so far on 4 different vCenters that doesn't have anything to do with each other but on one of them I cannot have that method working because I get "Caught vmodl fault : A specified parameter was not correct: spec.url" error, with the issue being in this line: "ticket = session_manager.AcquireGenericServiceTicket(service_request_spec)".

I didn't want to open this as an issue as it might be a known one already or maybe it has something to do with vcenter/esxi versions? I already tried to debug and checked all the settings and they are literally the same and I ran out of ideas. Even if I try to print it out service_request_spec it has the exactly same structure on all of them.

Is there any known issue? Or if you could have any suggestion on what could be wrong, please share.

PS: on all the other vCenters the method is working just fine.

**Comment by prziborowski** [comment, marked as answer]
You mean when you call: https://github.com/vmware/pyvmomi-community-samples/blob/87f49fe39fa48ea5cbf304550834b674afe0d3eb/samples/upload_file_to_datastore.py#L132 it is throwing a fault?
That call is commented out, so I'm not sure if the previous code in the file is supposed to work with it.

The rest of the file is using the mechanism that https://$VCIP/folder interface provides to download/upload.
The acquireGenericServiceTicket method is usually to access the ESXi host directly and download/upload.
If you want to use that method, then the URL you would pass (if you don't know the host to use and just want to provide datastore) is:
`ds:///vmfs/volumes/<uuid>/<path>?dc=<datacenter>`. The response from the method would be a URL that should include a hostname/IP address of ESX that you can httpPut the file with.
I don't think this is documented anywhere, which is probably why the author of upload_file_to_datastore.py didn't use it and commented out the method call.

**Reply by elungu3** [reply]
It's weird that it works after I commented that back, because the reason I uncommented it was because I was getting some Method Not Allowed/Permission Denied error. No ideea why, but I am glad its working tho. Thanks a lot!
